src/schema/v2/previewSavedSearch.ts

```diff
@@ -99,13 +101,72 @@ const PreviewSavedSearchType = new GraphQLObjectType<any, ResolverContext>({
       type: new GraphQLNonNull(
         new GraphQLList(new GraphQLNonNull(SearchCriteriaLabel))
       ),
-      resolve: (_) => {
-        return mockSuggestedFilters
+      resolve: async (_root, _args, { filterArtworksLoader }) => {
+        const suggestedFiltersByArtist: SearchCriteriaLabel[][] = await Promise.all(
+          _root.artistIDs.map(
+            async (artistSlug) =>
+              await getSuggestedFiltersByArtistSlug(
+                artistSlug,
+                filterArtworksLoader
+              )
+          )
+        )
+
+        return _.chain(suggestedFiltersByArtist)
+          .flatten()
+          .uniqBy((searchCriteria) => JSON.stringify(searchCriteria))
+          .value()
       },
     },
   }),
 })
 
+const getMostPopularField = (aggregation: {
+  [key: string]: { name: string; count: number }
+}): string => {
+  return Object.values(aggregation).reduce((acc, curr) => {
+    return curr.count > acc.count ? curr : acc
+  }).name
+}
+
+const getRaritySearchCriteriaLabel = (
+  name: string | undefined
+): SearchCriteriaLabel => {
+  if (!name) {
```

Comment:
Is this the case if there are no published artworks for the artist with any rarity associated?

Reply:
this is actually for the case where aggregations is empty - I can work around it maybe to make sure we don't get here

Reply:
Not sure if there could be other cases where aggregations are empty, but anyway, it's probably a good idea to return a default in this case 👍